File: app/src/main/java/com/tt/handsomeman/model/SignUpFormState.java
package com.tt.handsomeman.model;

import androidx.annotation.Nullable;

public class SignUpFormState {
    @Nullable
    private Integer nameError;
    @Nullable
    private Integer mailError;
    @Nullable
    private Integer passwordError;
    @Nullable
    private Integer rePasswordError;

    private boolean isDataValid;

    public SignUpFormState(@Nullable Integer nameError, @Nullable Integer mailError, @Nullable Integer passwordError, @Nullable Integer rePasswordError) {
        this.nameError = nameError;
        this.mailError = mailError;
        this.passwordError = passwordError;
        this.rePasswordError = rePasswordError;
        this.isDataValid = false;
    }

    public SignUpFormState(boolean isDataValid) {
        this.nameError = null;
        this.mailError = null;
        this.passwordError = null;
        this.rePasswordError = null;
        this.isDataValid = isDataValid;
    }

    @Nullable
    public Integer getNameError() {
        return nameError;
    }

    @Nullable
    public Integer getMailError() {
        return mailError;
    }

    @Nullable
    public Integer getPasswordError() {
        return passwordError;
    }

    @Nullable
    public Integer getRePasswordError() {
        return rePasswordError;
    }

    public boolean isDataValid() {
        return isDataValid;
    }
}

File: app/src/main/java/com/tt/handsomeman/model/SignUpFormState.kt
package com.tt.handsomeman.model

class SignUpFormState {
    var nameError: Int? = null
        private set
    var mailError: Int? = null
        private set
    var passwordError: Int? = null
        private set
    var rePasswordError: Int? = null
        private set

    var isDataValid: Boolean = false
        private set

    constructor(nameError: Int?, mailError: Int?, passwordError: Int?, rePasswordError: Int?) {
        this.nameError = nameError
        this.mailError = mailError
        this.passwordError = passwordError
        this.rePasswordError = rePasswordError
        this.isDataValid = false
    }

    constructor(isDataValid: Boolean) {
        this.nameError = null
        this.mailError = null
        this.passwordError = null
        this.rePasswordError = null
        this.isDataValid = isDataValid
    }
}
